internal/testing/enginetest/enginetest.go

```diff
@@ -212,21 +215,188 @@ func RunTestModuleEngine_Call_HostFn(t *testing.T, et EngineTester) {
 		require.Equal(t, uint64(3), results[0])
 		require.Same(t, memory, ctxMemory)
 	})
+}
 
-	t.Run("errs when not enough parameters", func(t *testing.T) {
-		_, err := me.Call(modCtx, f)
-		require.EqualError(t, err, "expected 1 params, but passed 0")
-	})
+func RunTestModuleEngine_Call_HostFn(t *testing.T, et EngineTester) {
+	runTestModuleEngine_Call_HostFn_ModuleContext(t, et) // TODO: refactor to use the same test interface.
 
-	t.Run("errs when too many parameters", func(t *testing.T) {
-		_, err := me.Call(modCtx, f, 1, 2)
-		require.EqualError(t, err, "expected 1 params, but passed 2")
-	})
+	e := et.NewEngine()
+
+	imported, importedMe, importing, importingMe := setupHostFunctions(t, e)
+	defer closeModuleEngineWithExitCode(t, importedMe, 0)
+	defer closeModuleEngineWithExitCode(t, importingMe, 0)
+
+	// Ensure the base case doesn't fail: A single parameter should work as that matches the function signature.
+	for me, fn := range map[wasm.ModuleEngine]*wasm.FunctionInstance{
+		importedMe:  imported.Exports[hostFnName].Function,
+		importedMe:  imported.Exports[callHostFnName].Function,
```

Comment:
same keys are used in the map literal and seems odd to me..?

Reply:
I'll switch it to a matrix test

Reply:
done in the other PR